### DISCUSSION TITLE
example of a test or a contract that demonstrates a stream that unlocks in steps?

### DISCUSSION THREAD

**Discussion by viraj124** [sablier-labs/evm-monorepo#690] [Q&A] [answered]
Hi,

Is there an example of a test or contract demonstrating this type of stream https://docs.sablier.com/concepts/protocol/stream-types#unlock-in-steps?

**Comment by PaulRBerg** [comment, marked as answer]
Hi @viraj124, thank you for your question.

Unfortunately, we do not have any hard-coded test for the unlock-in-steps payment model in Lockup Dynamic. However, this is a great (implicit) feature request. It would be helpful to provide such hard-coded tests for the Lockup Dynamic curves available in the UI. I have gone ahead and created [an issue](https://github.com/sablier-labs/examples/issues/16) to track the implementation for this.

In the meantime, there are two things that I recommend:

1. Look at the source code behind the "Types of Streams" page in our docs: https://github.com/sablier-labs/v2-docs/blob/d5c2be0b4463af7720a56927c6abe135527489a9/docs/concepts/protocol/02-stream-types.mdx#L120-L139
2. Take our [UI sandbox](https://github.com/sablier-labs/v2-sandbox) for a spin and experiment with different segment values there

**Reply by viraj124** [reply]
thanks a lot :)
I'll check it out

**Reply by PaulRBerg** [reply]
Hey @viraj124! We now have examples for Lockup Dynamic streams that unlock in steps:

https://github.com/sablier-labs/examples/blob/7a400ade04ae4fd9ea98bfcc7fae1ccd4011ace8/v2/core/LockupDynamicCurvesCreator.sol
